### PR TITLE
#20725 support YellowBrick procedures recognition

### LINKED ISSUE #20725: Cannot (easily) drop stored procedure in Yellowbrick
### Description

When using the GUI, attempting to right-click a stored procedure and choose "delete" will not work.
The script that is generated behind the scenes is not syntactically correct for yellowbrick.  The generated script uses "DROP FUNCTION..." but Yellowbrick expects "DROP PROCEDURE"

If I proceed with the incorrect script, I get the following error:
**SQL Error [42809]: ERROR: "blah.blah" is a stored procedure
  Hint: Use DROP PROCEDURE to drop stored procedures.**

### DBeaver Version

Community Edition 23.1.3

### Operating System

Windows 10 64 bit 22H2 build 19045

### Database and driver

Yellowbrick 5.2
Latest Postgres JDBC driver

### Steps to reproduce

1. right click any stored procedure in a YB database
2. choose "delete" from the context menu
3. In the "Delete Object" dialog, click the "View Script" button
4. Observe the incorrect syntax

### Additional context
Of course the workaround is to just write and execute a "DROP PROCEDURE" statement myself.

error log

image

### PR DESCRIPTION
**Pull request by LonwoLonwo**
2023-08-02 09_22_37-Delete script

You can check only YellowBrick. No impact on other forks.

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreServerExtension.java [hunk 1]
```diff
@@ -104,6 +104,11 @@ public interface PostgreServerExtension {
     String getProceduresSystemTable();
     String getProceduresOidColumn();
 
+    /**
+     * Returns true if a database contains prosp column in the pg_proc system table. This column knows about routine type.
+     */
+    boolean supportsProSPColumn();
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreServerExtension.java]: Lets not add yet another functin to ServerType. We can determine column presence in loadInfo (lookup in resultset metadata)

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreProcedure.java [hunk 2]
```diff
@@ -306,7 +306,17 @@ private void loadInfo(DBRProgressMonitor monitor, ResultSet dbResult) {
             } else if (isWindow) {
                 kind = PostgreProcedureKind.w;
             } else {
-                kind = PostgreProcedureKind.f;
+                boolean isProcedure = false;
+                try {
+                    isProcedure = dbResult.getBoolean("prosp");
+                } catch (SQLException e) {
+                    // Slip then. This column only persist in special cases
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreProcedure.java:313]:
```suggestion
                } catch (SQLException ignored) {
                    // Skip then. This column only persist in special cases
```

**Comment by Matvey16** [comment]
Verified

**Pull request merged**